### PR TITLE
Bulk metadata ARXIVNG-332

### PR DESCRIPTION
**Pull request by JaimieMurdock**

### CODE REVIEW

File: search/agent/consumer.py [hunk 1]
```diff
@@ -165,6 +166,79 @@ def _get_metadata(self, arxiv_id: str) -> DocMeta:
 
         return docmeta
 
+    def _get_bulk_metadata(self, arxiv_ids: List[str]) -> Dict[str,DocMeta]:
+        """
+        Retrieve metadata from the :mod:`.metadata` service for multiple documents.
+
+        Parameters
+        ----------
+        arxiv_id : str
```
Comment (erickpeirson) [inline search/agent/consumer.py]: Looks like the docstring could use an update

File: search/services/metadata.py [hunk 2]
```diff
@@ -142,6 +144,66 @@ def retrieve(self, document_id: str) -> DocMeta:
         return data
 
 
+    def bulk_retrieve(self, document_ids: List[str]) -> Dict[str, DocMeta]:
+        """
+        Retrieve metadata for an arXiv paper.
+
```
Comment (erickpeirson) [inline search/services/metadata.py]: This one too :-)

File: search/services/metadata.py [hunk 3]
```diff
@@ -142,6 +144,66 @@ def retrieve(self, document_id: str) -> DocMeta:
...
+            logger.error('SSLError: %s', e)
+            raise SecurityException('SSL failed: %s' % e) from e
+        except requests.exceptions.ConnectionError as e:
+            logger.error('ConnectionError: %s', e)
+            raise ConnectionFailed(
+                'Could not connect to metadata service: %s' % e
+            ) from e
+
+        if response.status_code not in \
+                [status.HTTP_200_OK, status.HTTP_206_PARTIAL_CONTENT]:
```
Comment (erickpeirson) [inline search/services/metadata.py]: Thinking out loud: what would actually happen if we got a 206? Would we risk not being able to deserialize the content?

Comment (erickpeirson) [reply search/services/metadata.py]: In which case it would be handled below, I suppose

Comment (mhl10) [reply search/services/metadata.py]: Also thinking out loud here: we should consider adding a parameter to cap the number of `document_ids` per request to the bulk_docmeta endpoint, similar to what id being done with bulk indexing.

**Pull request merged**